consensus/hotstuff/safetyrules/safety_rules_test.go

```diff
@@ -286,31 +286,20 @@ func (s *SafetyRulesTestSuite) TestProduceVote_PersistStateException() {
 
 // TestProduceVote_VotingOnUnsafeProposal tests different scenarios where we try to vote on unsafe blocks
 func (s *SafetyRulesTestSuite) TestProduceVote_VotingOnUnsafeProposal() {
```

Comment:
Would suggest a bit more documentation:
* for most test cases in `TestProduceVote_VotingOnUnsafeProposal`, we expect an exception. This is because all blocks are supposed to be pre-validated. And if any of the sanity checks fail, we interpret this as a symptom of an internal bug in the validation logic. 

### Suggestions:
1. Rename `TestProduceVote_VotingOnUnsafeProposal` to `TestProduceVote_VotingOnInvalidProposals` and include the following documentation
    ```suggestion
    // TestProduceVote_VotingOnInvalidProposals tests different scenarios where we try to vote on unsafe blocks
    // SafetyRules contain a variety of checks to confirm that QC and TC have the desired relationship to each other.
    // In particular, we test:
    //    (i) A TC should be included in a proposal, if and only of the QC is not the prior view.
    //   (ii) When the proposal includes a TC (i.e. the QC not being for the prior view), the TC must be for the prior view.
    //  (iii) The QC in the block must have a smaller view than the block.
    //   (iv) If the block contains a TC, the TC cannot contain a newer QC than the block itself.
    // Conditions (i) - (iv) are validity requirements for the block and all blocks that SafetyRules processes
    // are supposed to be pre-validated. Hence, failing any of those conditions means we have an internal bug.
    // Consequently, we expect SafetyRules to return exceptions but _not_ `NoVoteError`, because the latter 
    // indicates that the input block was valid but we didn't want to vote. 
    func (s *SafetyRulesTestSuite) TestProduceVote_VotingOnInvalidProposals() {
    ```
2 The only test scenarios in `TestProduceVote_VotingOnUnsafeProposal` that are different are the [last two cases](https://github.com/onflow/flow-go/blob/7ea4f828d37a5953bd4a4d1a01acbe422a8b6ebe/consensus/hotstuff/safetyrules/safety_rules_test.go#L371-L397)
   * regarding [`view-below-acknowledged`](https://github.com/onflow/flow-go/blob/7ea4f828d37a5953bd4a4d1a01acbe422a8b6ebe/consensus/hotstuff/safetyrules/safety_rules_test.go#L384-L397): I feel this test falls into the category of [`TestProduceVote_InvalidCurrentView`](https://github.com/onflow/flow-go/blob/7ea4f828d37a5953bd4a4d1a01acbe422a8b6ebe/consensus/hotstuff/safetyrules/safety_rules_test.go#L194-L203). 
       * Essentially we test the scenario where the `EventHandler` is feeding `SafetyRules` with wrong values for current view. In this particular case, we are testing that `SafetyRules` throws an exception if it receives inputs where values for `curView` are not monotonously increasing. 
       * I would suggest to also update the goDoc of [`TestProduceVote_InvalidCurrentView`](https://github.com/onflow/flow-go/blob/7ea4f828d37a5953bd4a4d1a01acbe422a8b6ebe/consensus/hotstuff/safetyrules/safety_rules_test.go#L194-L203) a bit (see my comment below)
   * regarding [`view-already-acknowledged`](https://github.com/onflow/flow-go/blob/7ea4f828d37a5953bd4a4d1a01acbe422a8b6ebe/consensus/hotstuff/safetyrules/safety_rules_test.go#L371-L383): I feel this is conceptually a different test case, that probably deserves its own test function:
      * Here we are testing that _only_ the first block for a view receives a vote (provided it is safe to vote). Other valid blocks imply that the proposer equivocated and should lead to a `NotVoteError`. 
      * I think this case is conceptually different compared to the other things we are testing in `TestProduceVote_VotingOnUnsafeProposal` is: here the input block by itself is actually valid; in contrast, the blocks for all the other cases are actually broken.